src/constraint.rs

```diff
@@ -345,3 +368,53 @@ impl ConstraintValidator for TypeConstraint {
         type_def.validate(value, type_store)
     }
 }
+
+/// Implements an `occurs` constraint of Ion Schema
+#[derive(Debug, Clone, PartialEq)]
+pub struct OccursConstraint {
+    isl_occurs: IslOccurs,
+}
+
+impl OccursConstraint {
+    pub fn new(isl_occurs: IslOccurs) -> Self {
+        Self { isl_occurs }
+    }
+}
+
+impl ConstraintValidator for OccursConstraint {
+    fn validate(&self, value: &OwnedElement, type_store: &TypeStore) -> ValidationResult {
+        todo!()
+    }
+}
+
+/// Implements an `ordered_elements` constraint of Ion Schema
+/// [ordered_elements]: https://amzn.github.io/ion-schema/docs/spec.html#all_of
+#[derive(Debug, Clone, PartialEq)]
+pub struct OrderedElementsConstraint {
+    type_ids: Vec<TypeId>,
+}
+
+impl OrderedElementsConstraint {
+    pub fn new(type_ids: Vec<TypeId>) -> Self {
+        Self { type_ids }
+    }
+
+    /// Tries to create an [OrderedElements] constraint from the given OwnedElement
+    pub fn resolve_from_isl_constraint(
+        type_references: &[IslTypeRef],
+        type_store: &mut TypeStore,
+        pending_types: &mut PendingTypes,
+    ) -> IonSchemaResult<Self> {
+        let resolved_types: Vec<TypeId> = type_references
+            .iter()
+            .map(|t| IslTypeRef::resolve_type_reference(t, type_store, pending_types))
+            .collect::<IonSchemaResult<Vec<TypeId>>>()?;
+        Ok(OrderedElementsConstraint::new(resolved_types))
+    }
+}
+
+impl ConstraintValidator for OrderedElementsConstraint {
+    fn validate(&self, value: &OwnedElement, type_store: &TypeStore) -> ValidationResult {
+        todo!()
```

Comment:
I assume you've left these as `todo!` intentionally. Consider:
1. opening an issue that you can add in a comment to track it
2. calling out the remaining work for this feature in the PR description

These will help future readers of this commit/PR to understand its intent even if you don't get to the next step for a while.